src/org/labkey/test/tests/component/GridPanelBaseTest.java

```diff
@@ -0,0 +1,151 @@
+        List<String> newSets = new ArrayList<>();
+
+        for (String allCombination : allSets)
+        {
+            newSets.add(allCombination + values.get(index));
+        }
+
+        allSets.addAll(newSets);
+
+        return allSets;
+    }
+
+    protected void initProject()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+
+        // Add the 'Sample Types' web part. It is easier when debugging etc...
+        PortalHelper portalHelper = new PortalHelper(this);
+        portalHelper.enterAdminMode();
+        portalHelper.addWebPart("Sample Types");
+        portalHelper.exitAdminMode();
+    }
+
+    /**
+     * Helper to remove the 'Flag' column from the default view. It just gets in the way for some tests, and is easier
+     * to remove it.
+     *
+     * @param sampleType Name of sample type.
+     */
+    protected void removeFlagColumnFromDefaultView(String sampleType)
+    {
+        goToProjectHome();
+
+        refresh();
+
+        waitAndClickAndWait(Locator.linkWithText(sampleType));
+
+        log(String.format("Remove '%s' column form default view.", REMOVED_FLAG_COLUMN));
+
+        SampleTypeHelper sampleHelper = new SampleTypeHelper(this);
+        DataRegionTable drtSamples = sampleHelper.getSamplesDataRegionTable();
+        drtSamples.goToView(VIEW_DEFAULT);
+
+        CustomizeView cv = drtSamples.openCustomizeGrid();
+        cv.removeColumn(REMOVED_FLAG_COLUMN);
+
+        // This will save the changes to the default view for everyone. The view menu option will remain as "Default"
+        cv.saveCustomView("", true);
+
+    }
+
+    /**
+     * Go to core-components.view page for the given project. Then select the QueryGrid (GridPanel) component. Clear any
+     * selections, filters or search values that are present.
+     *
+     * @param sampleType The sample type used to populate the grid.
+     * @return A queryGrid object.
+     */
+    protected QueryGrid beginAtQueryGrid(String sampleType)
+    {
+        return beginAtQueryGrid(sampleType, true);
+    }
+
+    /**
+     * Go to core-components.view page for the given project. Then select the QueryGrid (GridPanel) component.
+     *
+     * @param sampleType The sample type used to populate the grid.
+     * @param clearFilters Clear any filters. If the default view has been changed to use a filter the filter pill will
+     *                     be shown. Setting this to true will clear the filter, and as a result will mark the current
+     *                     default view as changed.
+     * @return A queryGrid object.
+     */
+    protected QueryGrid beginAtQueryGrid(String sampleType, boolean clearFilters)
+    {
+        QueryGrid grid = CoreComponentsTestPage.beginAt(this, getProjectName())
+                .getGridPanel(TEST_SCHEMA, sampleType);
+
+        grid.clearAllSelections();
+
+        if(clearFilters)
+        {
+            grid.clearFilters();
+        }
+
+        grid.clearSearch();
```

Comment:
Consider adding a `resetView` method to `QueryGrid` that does all of this. This method has no way to deal with a combination of custom filters and view filters.
```suggestion
        grid.resetView();
```
In `QueryGrid`:
```
    public void resetView()
    {
        clearAllSelections();

        if(isUndoButtonVisible())
        {
            clickUndoButton();
        }

       clearSearch();
    }
```

Reply:
I see your point, but can you think of a better method name? Now that views are available in this grid, and you can edit an existing view, a method named `resetView` could be misinterpreted.
Maybe something more verbose like `resetDefaultClearSelections()`?

Reply:
Oh, certainly. I wasn't attached to the name; just thought it would be useful functionality since we frequently want to just reset a grid.

Reply:
I've `added QueryGrid.resetToDefaultState(boolean revertDefaultView)`.
It will do the best it can to get the grid back to its original state. It will apply the default view. If the parameter is set to true it will revert the default view before applying it, otherwise it will apply the default view with any modifications it may have.